### ISSUE TITLE
loss function vs objective function

### ISSUE THREAD

**Issue by germayneng** [dmlc/xgboost#2862]
Good day,

This is a question more in theory than for practical. I am currently trying to understand gradient boosting as well as XGB in general. For gradient boosting and XGB, I understand that the loss function must be differentiable. I am currently confused what this loss function refers to. 

Another thing i want to clarify is that where does the eval_metric comes into play since we are working towards optimizing the loss function
